pagefaultgames/pokerogue
# Let Corrosion mons get badly poisoned by Toxic Orb

## What are the changes?

Toxic and Flame Orb use the holder's abilities when trying to set status.
In the main series, Toxic Orb will activate on a Poison or Steel type with the ability Corrosion and inflict bad poison. Here, since there's a possibility of two abilities on the same mon, it also allows abilities to interact. For example, an Axew with Mold Breaker ability and Immunity passive would become badly poisoned by Toxic Orb. It would be as if an opponent with Mold Breaker used Toxic on it, essentially.

## Why am I doing these changes?



The way these orbs work in the official games considers the item holder as the "source" of the status condition. This is a very situational but also very funny edge case, and I'm sure other items have conditions similar to this we could look at later on. The Mold Breaker/Immunity interaction doesn't exist in the main games, but is likely how it would work if there ever were a way for one mon to have two abilities. It's consistent with the Corrosion interaction.

## What did change?


This PR simply passes the item holder as the "source" of the status condition for Toxic and Flame Orb. This would allow the proper abilities to apply, such as Corrosion on a Poison- or Steel-type, or Mold Breaker with a status immunity ability. 

### Screenshots/Videos


Bulbasaur, who does not have Corrosion, is not poisoned by Toxic Orb.
![image](https://github.com/pagefaultgames/pokerogue/assets/36677462/fa868862-ff29-4ff8-8e65-253104acf6bb)
Salandit, who does have Corrosion, is.
![image](https://github.com/pagefaultgames/pokerogue/assets/36677462/d241bc4c-7d26-482b-904f-c3d093b4d6ca)
Magnemite, who I gave Corrosion as a passive via override, is.
![image](https://github.com/pagefaultgames/pokerogue/assets/36677462/5f2c25cc-d7ff-4fd0-be27-68d9ba2e6e46)
It doesn't get poisoned if it's immune due to ability such as Immunity (also override).
![image](https://github.com/pagefaultgames/pokerogue/assets/36677462/8c21cbdd-afcc-4998-81a7-0c9344767b3d)
Salandit with Neutralizing Gas passive will still Poison itself...
![image](https://github.com/pagefaultgames/pokerogue/assets/36677462/aa5582f3-aded-49ab-8de6-7b5d668f46bd)
But that's just because Neutralizing Gas doesn't work on Corrosion.
![image](https://github.com/pagefaultgames/pokerogue/assets/36677462/aa6159f3-ffc0-417e-b26a-2a4aa0bd1c47)
It otherwise seems to work as expected.
![image](https://github.com/pagefaultgames/pokerogue/assets/36677462/29900d5e-c69f-45a1-8533-28c6cfc50d11)
Something to note is that a mon with both Mold Breaker and Immunity (due to passives) will also poison itself from Toxic Orb.
![image](https://github.com/pagefaultgames/pokerogue/assets/36677462/2d8625fc-8835-47c9-8c93-abccf5a725ca)


## How to test the changes?




- Add `{"name":"TOXIC_ORB"}` to the starting held items overrides
- Start a game with Corrosion Salandit or Glimmet to see it work
- Start a game with another Poison- or Steel-type mon to see it not work
- Start a game with a Mold Breaker mon and set the passive ability override to Immunity to see that interaction

## Checklist
